atmos_model.F90

```diff
@@ -2337,932 +2353,252 @@ subroutine setup_exportdata()
+            case ('v')
+              call block_atmos_copy(datar83d, Atm(mygrid)%v, Atm_block, nb)
+            case ('ua')
+              call block_atmos_copy(datar83d, Atm(mygrid)%ua, Atm_block, nb)
+            case ('va')
+              call block_atmos_copy(datar83d, Atm(mygrid)%va, Atm_block, nb)
+            case ('t')
+              call block_atmos_copy(datar83d, Atm(mygrid)%pt, Atm_block, nb)
+            case ('delp')
+              call block_atmos_copy(datar83d, Atm(mygrid)%delp, Atm_block, nb)
+            case ('sphum')
+              sphum = get_tracer_index(MODEL_ATMOS, 'sphum')
+              call block_atmos_copy(datar83d, Atm(mygrid)%q, sphum, Atm_block, nb)
+            case ('ice_wat')
+              ice_wat = get_tracer_index(MODEL_ATMOS, 'ice_wat')
+              call block_atmos_copy(datar83d, Atm(mygrid)%q, ice_wat, Atm_block, nb)
+            case ('liq_wat')
+              liq_wat = get_tracer_index(MODEL_ATMOS, 'liq_wat')
+              call block_atmos_copy(datar83d, Atm(mygrid)%q, liq_wat, Atm_block, nb)
+            case ('o3mr')
+              o3mr = get_tracer_index(MODEL_ATMOS, 'o3mr')
+              call block_atmos_copy(datar83d, Atm(mygrid)%q, o3mr, Atm_block, nb)
+            case ('phis')
+              call block_atmos_copy(datar82d, Atm(mygrid)%phis, Atm_block, nb)
+            case ('u_srf')
+              call block_atmos_copy(datar82d, Atm(mygrid)%u_srf, Atm_block, nb)
+            case ('v_srf')
+              call block_atmos_copy(datar82d, Atm(mygrid)%v_srf, Atm_block, nb)
+            case ('slmsk')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%slmsk, Atm_block, nb)
+            case ('weasd')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%weasd, Atm_block, nb)
+            case ('tsea')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%tsfco, Atm_block, nb)
+            case ('vtype')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vtype, Atm_block, nb)
+            case ('stype')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%stype, Atm_block, nb)
+            case ('vfrac')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%vfrac, Atm_block, nb)
+            case ('stc')
+              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%stc, Atm_block, nb)
+            case ('smc')
+              call block_data_copy(datar83d, GFS_data(nb)%sfcprop%smc, Atm_block, nb)
+            case ('snwdph')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%snowd, Atm_block, nb)
+            case ('f10m')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%f10m, Atm_block, nb)
+            case ('zorl')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%zorl, Atm_block, nb)
+            case ('t2m')
+              call block_data_copy(datar82d, GFS_data(nb)%sfcprop%t2m, Atm_block, nb)
+          end select
```

Comment:
Should we have a default case with a warning message or an error?

Reply:
Since the coupling fields may or may not be available, there is no reference configuration we can use to issue warnings.
The generic subroutines check if both source and destination pointers are associated before attempting to copy the data to prevent cases where not all listed fields are connected (which would crash the model).

Reply:
If the field is realized, that means FV3 advertised it and some other component is expecting 'valid' values. If we can not provide the values it is either logic error or a bug. In either case we should at least have some warning, in not an error.